src/editor/Core/Types.re

```diff
@@ -58,6 +58,37 @@ module BufferPosition = {
   };
 };
 
+module ViewType = {
```

Comment:
My thinking here is that by adding a Variant to each type of view element then down the line if you were filtering a list that contained multiple types you could pattern match to group them.

I was specifically thinking that if we pursue the route of showing tabs and buffers in a similar way to [vimWinTabs](https://github.com/zefei/vim-wintabs) it might be useful to be able to group these :man_shrugging:

Reply:
I removed this as I thought it might prove a point of contention 😆 can discuss further when/if its actually necessary